packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -0,0 +1,373 @@
+  final MaterialStateProperty<Color?>? surfaceTintColor;
+
+  /// The highlight color that's typically used to indicate that
+  /// the button is focused, hovered, or pressed.
+  final MaterialStateProperty<Color?>? overlayColor;
+
+  /// The color and weight of the search bar's outline.
+  ///
+  /// This value is combined with [shape] to create a shape decorated
+  /// with an outline.
+  ///
+  /// If null, the value of [SearchBarThemeData.side] will be used. If this is
+  /// also null, the search bar doesn't have a side by default.
+  final MaterialStateProperty<BorderSide?>? side;
+
+  /// The shape of the search bar's underlying [Material].
+  ///
+  /// This shape is combined with [side] to create a shape decorated
+  /// with an outline.
+  ///
+  /// If null, the value of [SearchBarThemeData.shape] will be used.
+  /// If this is also null, then the default value is 16.0 horizontally.
+  /// Defaults to [StadiumBorder].
+  final MaterialStateProperty<OutlinedBorder?>? shape;
+
+  /// The padding between the search bar's boundary and its child.
+  ///
+  /// If null, the value of [SearchBarThemeData.padding] will be used.
+  /// If this is also null, then the default value is 16.0 horizontally.
+  final MaterialStateProperty<EdgeInsetsGeometry?>? padding;
+
+  /// The style to use for the text being edited.
+  ///
+  /// If null, defaults to the `bodyLarge` text style from the current [Theme].
+  /// The default text color is [ColorScheme.onSurface].
+  final MaterialStateProperty<TextStyle?>? textStyle;
+
+  /// The style to use for the [hintText].
+  ///
+  /// If null, the value of [SearchBarThemeData.hintStyle] will be used. If this
+  /// is also null, the value of [textStyle] will be used. If this is also null,
+  /// defaults to the `bodyLarge` text style from the current [Theme].
+  /// The default text color is [ColorScheme.onSurfaceVariant].
+  final MaterialStateProperty<TextStyle?>? hintStyle;
+
+  @override
+  State<SearchBar> createState() => _SearchBarState();
+}
+
+class _SearchBarState extends State<SearchBar> {
+  late final MaterialStatesController _internalStatesController;
+  late final FocusNode _focusNode;
+
+  @override
+  void initState() {
+    super.initState();
+    _internalStatesController = MaterialStatesController();
+    _internalStatesController.addListener(() {
+      setState(() {});
+    });
+    _focusNode = widget.focusNode ?? FocusNode();
+  }
+
+  @override
+  void dispose() {
+    _internalStatesController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final TextDirection textDirection = Directionality.of(context);
+    final ColorScheme colorScheme = Theme.of(context).colorScheme;
+    final IconThemeData iconTheme = IconTheme.of(context);
+    final SearchBarThemeData searchBarTheme = SearchBarTheme.of(context);
+    final SearchBarThemeData defaults = _SearchBarDefaultsM3(context);
+
+    T? resolve<T>(MaterialStateProperty<T>? property) {
```

Comment:
Unfortunately this isn't quite right. For example, if widget.textStyle is non-null but it resolves to null, then we should defer searchBarTheme.textStyle.

This function should be defined roughly like this:
```dart
T? resolve<T>(
  MaterialStateProperty<T>? widgetValue,
  MaterialStateProperty<T>? themeValue,
  MaterialStateProperty<T>? defaultValue,
) {
  final Set<MaterialState> states = _internalStatesController.value;
  return widgetValue?.resolve(states) ?? themeValue?.resolve(states) ?? defaultValue.resolve(states);
}
```

Reply:
Thanks a lot for catching this! Just fixed it.